chalice/app.py

```diff
@@ -266,15 +274,29 @@ def __init__(self, body, headers=None, status_code=200):
         self.headers = headers
         self.status_code = status_code
 
-    def to_dict(self):
+    def _base64encode(self, data):
+        if not isinstance(data, bytes):
+            raise ValueError('Expected bytes type for body with binary '
+                             'Content-Type. Got %s type body instead.'
+                             % type(data))
+        data = base64.b64encode(data)
+        return data.decode('ascii')
+
+    def to_dict(self, binary_types=None):
+        if binary_types is None:
+            binary_types = []
         body = self.body
-        if not isinstance(body, str):
-            body = json.dumps(body, default=handle_decimals)
-        return {
+        response = {
             'headers': self.headers,
             'statusCode': self.status_code,
-            'body': body,
         }
+        if self.headers.get('Content-Type') in binary_types:
```

Comment:
I think we should use the case insensitive mapping like we do for incoming requests.  Also we should think about smarter content type matching.  There were a few bugs related to that for the content type matching on receiving a request in `Chalice.__call__` (stuff about charsets I believe).  Ideally we reuse the same logic throughout the code base for content type matching